nominal/core/asset.py

```diff
@@ -229,6 +230,36 @@ def list_data_scopes(self) -> Sequence[tuple[str, ScopeType]]:
         """
         return (*self.list_datasets(), *self.list_connections(), *self._list_logsets(), *self.list_videos())
 
+    def get_or_create_dataset(
+        self,
+        *,
+        data_scope_name: str,
+        name: str | None = None,
+        description: str | None = None,
+        labels: Sequence[str] = (),
+        properties: Mapping[str, str] | None = None,
+    ) -> Dataset:
+        """Retrieve a dataset by data scope name, or create a new one if it does not exist."""
+        try:
+            return self.get_dataset(data_scope_name)
+        except ValueError:
```

Comment:
we should probably make get_dataset return a more nominal-specific error rather than a value error...but that doesn't have to be in this PR